packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -258,26 +258,30 @@ export class GadgetConnection {
   }
 
   private newBaseClient() {
-    return new Client({
-      url: this.endpoint,
-      fetch: this.fetch,
-      exchanges: [
-        dedupExchange,
-        multipartFetchExchange,
-        subscriptionExchange({
-          forwardSubscription: (operation) => {
-            return {
-              subscribe: (sink) => {
-                const dispose = this.baseSubscriptionClient.subscribe(operation, sink as Sink<ExecutionResult>);
-                return {
-                  unsubscribe: dispose,
-                };
-              },
-            };
-          },
-        }),
-      ],
-    });
+    const exchanges = [dedupExchange];
+
+    // apply urql's default caching behaviour when client side (but skip it server side)
+    if (typeof window != "undefined") {
+      exchanges.push(cacheExchange);
```

Comment:
So reading up on this a while back, I think we need to update everything to always add `__typename` and `id` to the query to make this valuable?

Reply:
We are already woop woop! https://github.com/gadget-inc/js-clients/blob/2afa56e477231acdd1c5eb96e94cb978e9f7563b/packages/api-client-core/src/FieldSelection.ts#L20 is a thing which the client sets most of the time.

Reply:
Ah yes, I see it defaults to `false`, but we generally use `true`. How about the id? Or do you think it's a "use id to make it fast and good, but your choice to not select it?" kind of dealio?

Reply:
I think it would be wise to fetch it, but, document caching doesn't actually need it so I don't think we should force it. If we add graph-cache support then it'd be a thing at that point for sure.

Reply:
Oh right, my bad. I was thinking this was graph cache 😅